trino/auth.py

```diff
@@ -377,53 +381,53 @@ def _get_token(self, token_server, response, **kwargs):
 
         raise exceptions.TrinoAuthError("Exceeded max attempts while getting the token")
 
-    def _get_token_from_cache(self, host: str) -> Optional[str]:
+    def _get_token_from_cache(self, host: Optional[str]) -> Optional[str]:
         with self._token_lock:
             return self._token_cache.get_token_from_cache(host)
 
-    def _store_token_to_cache(self, host: str, token: str) -> None:
+    def _store_token_to_cache(self, host: Optional[str], token: str) -> None:
         with self._token_lock:
             self._token_cache.store_token_to_cache(host, token)
 
     @staticmethod
-    def _determine_host(url) -> Optional[str]:
+    def _determine_host(url: Optional[str]) -> Any:
         return urlparse(url).hostname
 
 
 class OAuth2Authentication(Authentication):
-    def __init__(self, redirect_auth_url_handler=CompositeRedirectHandler([
+    def __init__(self, redirect_auth_url_handler: CompositeRedirectHandler = CompositeRedirectHandler([
         WebBrowserRedirectHandler(),
         ConsoleRedirectHandler()
     ])):
         self._redirect_auth_url = redirect_auth_url_handler
         self._bearer = _OAuth2TokenBearer(self._redirect_auth_url)
 
-    def set_http_session(self, http_session):
+    def set_http_session(self, http_session: Session) -> Session:
         http_session.auth = self._bearer
         return http_session
 
-    def get_exceptions(self):
+    def get_exceptions(self) -> Tuple[Any, ...]:
         return ()
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         if not isinstance(other, OAuth2Authentication):
             return False
         return self._redirect_auth_url == other._redirect_auth_url
 
 
 class CertificateAuthentication(Authentication):
-    def __init__(self, cert, key):
+    def __init__(self, cert: str, key: str):
         self._cert = cert
         self._key = key
 
-    def set_http_session(self, http_session):
+    def set_http_session(self, http_session: Session) -> Session:
         http_session.cert = (self._cert, self._key)
         return http_session
 
-    def get_exceptions(self):
+    def get_exceptions(self) -> Tuple[Any, ...]:
         return ()
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```

Comment:
`other: Authentication`?

Reply:
object is actually correct as equality can check against whatever other Python type (so `object`).